client/src/index.jsx

```diff
@@ -34,36 +33,53 @@ export default class App extends Component {
   constructor( props ) {
     super( props );
     this.state = {
+      adults: 1,
+      children: 0,
+      infants: 0,
+      totalGuests: 1,
+      listingData: null,
     };
   }
 
   componentDidMount() {
-    this.getPathname().then( () => {
-      this.getListingData();
+    this.getListingData();
+  }
+
+  onGuestButtonClick = ( guest, increment ) => {
+    const val = this.state[ guest ] + increment;
+    const { totalGuests } = this.state;
+    const total = totalGuests + increment;
+    const { maxGuests } = this.state.listingData;
+    // infants don't count towards the number of guests, but have a cap of 5
+    if ( guest === 'infants' && ( val > 5 || val < 0 ) ) {
+      return;
+    // there is a minimum of 1 adult guest
+    } else if ( guest === 'adults' && ( val > maxGuests || val < 1 || total > maxGuests ) ) {
+      return;
+    } else if ( guest === 'children' && ( val > maxGuests || val < 0 || total > maxGuests ) ) {
+      return;
+    }
+    this.setState( {
+      [ guest ]: val,
+      totalGuests: guest === 'infants' ? totalGuests : total,
     } );
   }
 
-  getPathname = () => new Promise( ( resolve ) => {
+  getListingData = () => {
     let id = window.location.pathname;
     if ( id === '/' ) {
       id = 1;
     } else {
       id = id.replace( /\//g, '' );
     }
-    resolve( this.setState( {
-      pathname: id,
-    } ) );
-  } )
-
-  getListingData = () => {
-    axios.get( `http://127.0.0.1:3002/rooms/${ this.state.pathname }/bookingInfo/` )
+    axios.get( `http://127.0.0.1:3002/rooms/${ id }/bookingInfo/` )
       .then( ( response ) => {
         this.setState( {
           listingData: response.data,
         } );
       } )
-      .catch( ( error ) => {
-        console.log( error );
+      .catch( () => {
+        console.log( 'there was an error!' );
```

Comment:
If you are logging an error you might want to consider using console.error(), it will make your error much more noticeable in the console.

Reply:
In the end, I want to replace this with something to display to the user.